src/test/regress/upgrade/config.py

```diff
@@ -1,68 +1,297 @@
+        self.user = SUPER_USER_NAME
+        self.mixed_mode = arguments["--mixed"]
+        self.fixed_port = 57635
 
-WORKER_PORTS = [WORKER1PORT, WORKER2PORT]
-NODE_PORTS = {
-    COORDINATOR_NAME: COORDINATOR_PORT,
-    WORKER1: WORKER1PORT,
-    WORKER2: WORKER2PORT,
-}
+
+class PostgresConfig(CitusDefaultClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 0
+        self.is_citus = False
+
+
+class CitusSingleNodeClusterConfig(CitusDefaultClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 0
+
+
+class CitusSingleWorkerClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.worker_amount = 1
+
+
+class CitusSuperUserDefaultClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.user = SUPER_USER_NAME
+
+
+class CitusThreeWorkersManyShardsClusterConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {"citus.shard_count": 191}
+        self.worker_amount = 3
+
+    def setup_steps(self):
+        common.coordinator_should_haveshards(self.bindir, self.coordinator_port())
+
+
+class CitusSmallSharedPoolSizeConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {
+            # can be uncommented after https://github.com/citusdata/citus/issues/5342
+            # "citus.local_shared_pool_size": 5,
+            "citus.max_shared_pool_size": 5,
+        }
+
+
+class CitusSmallExecutorPoolSizeConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {
+            "citus.max_adaptive_executor_pool_size": 2,
+        }
+
+
+class CitusSequentialExecutionConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {
+            "citus.multi_shard_modify_mode": "sequential",
+        }
+
+
+class CitusCacheManyConnectionsConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {
+            "citus.max_cached_conns_per_worker": 4,
+        }
+
+
+class CitusUnusualExecutorConfig(CitusMXBaseClusterConfig):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self.new_settings = {
+            "citus.max_adaptive_executor_pool_size": 7,
+            "citus.executor_slow_start_interval": 1,
+            "citus.prevent_incomplete_connection_establishment": False,
+            "citus.enable_cost_based_connection_establishment": False,
+            "citus.max_cached_connection_lifetime": "10ms",
+            "citus.force_max_query_parallelization": "on",
```

Comment:
Because of https://github.com/citusdata/citus/issues/5345 we cannot use `force_max_query_parallelization`.

Can you add a comment here and link to the issue & comment out `force_max_query_parallelization to on`?